widget/entry.go

```diff
@@ -1136,10 +1171,9 @@ func (r *passwordRevealer) Cursor() desktop.Cursor {
 
 // Tapped satisfies the fyne.Tappable interface.
 func (r *passwordRevealer) Tapped(*fyne.PointEvent) {
-	r.entry.Lock()
-	r.entry.Password = !r.entry.Password
-	r.entry.Unlock()
-	r.Refresh()
+	r.entry.setFieldsAndRefresh(func() {
```

Comment:
Should this refresh here, doesn't 1177 call Focus which calls Refresh?

Reply:
Focus does not necessarily call refresh.
It might if the state has changed, but it's not part of the contract.
Here we are probably updating the display when we are already focused, so it's a driver detail whether or not the entry will refresh.